src/figuresClasses.ts

```diff
@@ -1,11 +1,82 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: 'triangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Invalid triangle: All sides must have non-zero length');
+    }
+
+    const longestSide = Math.max(a, b, c);
+    const sumTwoOthers = a + b + c - longestSide;
+
+    if (longestSide >= sumTwoOthers) {
+      throw new Error(
+        `Invalid triangle:
+        The longest side of a triangle is >= than a sum of two others`,
+      );
+    }
+
+    this.shape = 'triangle';
+  }
+
+  public getArea(): number {
+    const s = (1 / 2) * (this.a + this.b + this.c);
+    const square = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return +square.toFixed(2);
+  }
+}
+
+export class Circle implements Figure {
+  public shape: 'circle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public radius: number,
```

Comment:
[CODE KNOWLEDGE] - Similar to the Triangle class, the Circle class constructor doesn't need to manually assign 'this.color' and 'this.radius'. TypeScript handles this automatically when using 'public' in the constructor parameters.